### DISCUSSION TITLE
Plan for upgrading faiss-gpu

### DISCUSSION THREAD

**Discussion by yeoedward** [open-compass/opencompass#355] [View all discussions]
At the moment, OpenCompass depends on `faiss-gpu==1.7.2`. While newer versions of that package exist, they will no longer be released on PyPI because of package size limitations: https://github.com/kyamagu/faiss-wheels/issues/83

This means that as long as OpenCompass depends on the PyPI package `faiss-gpu`, it will not support python > 3.10 and also can't incorporate any upstream updates to faiss. I realize this might not be a major problem, but I was wondering if you guys have thought of any long term solutions to this.

I hit this limitation because I tried installing OpenCompass in a python 3.11 environment, which results in a `faiss-gpu` not found dependency error.

**Comment by tonysy** [comment]
Thanks for the insightful suggestion, we will add this into our backlog.
